File: RepoProbaParticipant.java
package me.repository;

import me.model.Participant;
import me.model.Proba;
import me.model.ProbaParticipant;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoProbaParticipant implements IRepoProbaParticipant {
    private static final Logger logger = LogManager.getLogger();
    JdbcUtils jdbcUtils;

    public RepoProbaParticipant(Properties props){
        logger.info("Initializing RepoAngajat with properties: {} ",props);
        jdbcUtils=new JdbcUtils(props);
    }

    @Override
    public int numarParticipantiProba(Integer p) { return 0; }

    @Override
    public List<ProbaParticipant> findProbaParticipantDupaProba(Integer p) {
        Connection con=jdbcUtils.getConnection();
        List<ProbaParticipant> probeparticipanti = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from ProbeParticipanti where idProba=?")){
            preStmt.setInt(1,p);
            try(ResultSet result=preStmt.executeQuery()) {
                while (result.next()) {
                    int idParticipant = result.getInt("idParticipant");
                    probeparticipanti.add(new ProbaParticipant(p,idParticipant));
                }
                return probeparticipanti;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public List<ProbaParticipant> findProbaParticipantDupaParticipant(Integer idParticipant) {
        Connection con=jdbcUtils.getConnection();
        List<ProbaParticipant> probeparticipanti = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from ProbeParticipanti where idParticipant=?")){
            preStmt.setInt(1,idParticipant);
            try(ResultSet result=preStmt.executeQuery()) {
                while (result.next()) {
                    int idProba = result.getInt("idProba");
                    probeparticipanti.add(new ProbaParticipant(idProba,idParticipant));
                }
                return probeparticipanti;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public ProbaParticipant save(ProbaParticipant elem) {
        logger.info("Se conecteaza la baza de date");
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into ProbeParticipanti (idProba,idParticipant) values (?,?)")){
            preStmt.setInt(1,elem.getIdProba());
            preStmt.setInt(2,elem.getIdParticipant());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.info("eroare baza de date: "+ex.getMessage());
        }
        logger.info("S-a adaugat in baza de date probaparticipant "+elem);
        return elem;
    }

    @Override
    public ProbaParticipant update(ProbaParticipant elem) { return null; }

    @Override
    public ProbaParticipant delete(ProbaParticipant elem) throws IllegalArgumentException {
        return null;
    }

    @Override
    public ProbaParticipant get(String s) { return null; }

    @Override
    public int size() { return 0; }

    @Override
    public Iterable<ProbaParticipant> findAll() { return null; }
}

File: Concurs Inot C#/RepoProbaParticipant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using log4net;

namespace Problema_3_CSharp.me.repository
{
    
    public class RepoProbaParticipant : IRepoProbaParticipant
    {
        private static readonly ILog log = LogManager.GetLogger("LoggerRepoProbaParticipant");
        public ProbaParticipant Delete(ProbaParticipant elem)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<ProbaParticipant> FindAll()
        {
            throw new NotImplementedException();
        }

        public IEnumerable<ProbaParticipant> FindProbaParticipantDupaParticipant(int p)
        {
            log.Info("Se incepe conectarea la bd si selectarea probe participantilor  " + p);
            var con = DBUtils.getConnection();
            IList<ProbaParticipant> probeparticipanti = new List<ProbaParticipant>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from ProbeParticipanti where idParticipant=@idParticipant";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@idParticipant";
                paramNume.Value = p;
                comm.Parameters.Add(paramNume);

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        ProbaParticipant pp = new ProbaParticipant(id,p);
                        probeparticipanti.Add(pp);
                    }
                }
            }
            return probeparticipanti;
        }

        public IEnumerable<ProbaParticipant> FindProbaParticipantDupaProba(int p)
        {
            log.Info("Se incepe conectarea la bd si selectarea probe participantilor  " + p);
            var con = DBUtils.getConnection();
            IList<ProbaParticipant> probeparticipanti = new List<ProbaParticipant>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from ProbeParticipanti where idProba=@idProba";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@idProba";
                paramNume.Value = p;
                comm.Parameters.Add(paramNume);

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(1);
                        ProbaParticipant pp = new ProbaParticipant(p,id);
                        probeparticipanti.Add(pp);
                    }
                }
            }
            return probeparticipanti;
        }

        public ProbaParticipant Get(string id)
        {
            throw new NotImplementedException();
        }

        public int NumarParticipantiProba(Proba p)
        {
            throw new NotImplementedException();
        }

        public ProbaParticipant Save(ProbaParticipant elem)
        {
            log.Info("Se incearca adaugarea");
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {

                comm.CommandText = "insert into ProbeParticipanti (idProba,idParticipant)  values (@idProba, @idParticipant)";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@idProba";
                paramNume.Value = elem.IdProba;
                comm.Parameters.Add(paramNume);

                var paramVarsta = comm.CreateParameter();
                paramVarsta.ParameterName = "@idParticipant";
                paramVarsta.Value = elem.IdParticipant;
                comm.Parameters.Add(paramVarsta);

                var result = comm.ExecuteNonQuery();

                log.Info("S-a adaugat " + elem.ToString());
                return elem;
            }
        }

        public int Size()
        {
            throw new NotImplementedException();
        }

        public ProbaParticipant Update(ProbaParticipant elem)
        {
            throw new NotImplementedException();
        }
    }
}
